### PR TITLE
Media player visual improvements

### LINKED ISSUE #6839: Escape key to dismiss Media Browser dialog
## The request

It would be nice if the escape key (and/or clicking outside the dialog) dismissed the media browser dialog. I think in many cases, you'll want to quickly select a song/app/movie or something and then close it after you make your selection. This should be quick to do imo.
.

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

1. Added a bit of space between media icon and its title
image

2. Added tooltip to opening button of media player selection
image

<s>3. Improved visuals if no media items were found (list with "No items" as text). Previously it was just a pure text without any styling.</s> (already tackled via https://github.com/home-assistant/frontend/pull/6813)
image

4. Make media player selection dialog close on outside klick and Escape key press.

5. Make media play dialog close on outside klick and Escape key press (fixes https://github.com/home-assistant/frontend/issues/6839).

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6839
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Please add a screenshot when making visual changes

**Comment by spacegaier** [comment]
> 
> 
> Please add a screenshot when making visual changes

Added to the PR description.

**Comment by zsarnett** [comment]
This will now need to be rebased

**Comment by spacegaier** [comment]
@zsarnett Rebase is done.

File: src/panels/media-browser/ha-panel-media-browser.ts [hunk 1]
```diff
@@ -64,14 +65,21 @@ class PanelMediaBrowser extends LitElement {
...
+                )}
+                @click=${this._showSelectMediaPlayerDialog}
+              >
+                <ha-svg-icon .path=${mdiPlayNetwork}></ha-svg-icon>
+              </mwc-icon-button>
+              <paper-tooltip animation-delay="0" position="left">
+                ${this.hass.localize(
+                  "ui.components.media-browser.choose_player"
+                )}
+              </paper-tooltip>
```
Comment (bramkragten) [inline src/panels/media-browser/ha-panel-media-browser.ts:81]: Let's not do this, a header button should not have a tooltip. It could have a title, and if it is not clear enough let's make it a regular button instead of an icon button.

File: src/components/media-player/ha-media-player-browse.ts [hunk 2]
```diff
@@ -726,6 +726,10 @@ export class HaMediaPlayerBrowse extends LitElement {
           background-color: transparent;
         }
 
+        mwc-list-item .title {
+          margin-left: 10px;
+        }
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:731]: Why is this needed? `mwc-list-item` should handle styling itself.

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: Without it, the song title is directly next to the song icon, which doesn't look good. With that change there is a bit of space (see screenshot).

File: src/panels/media-browser/ha-panel-media-browser.ts (hunk 1 above)
Comment (spacegaier) [reply src/panels/media-browser/ha-panel-media-browser.ts]: Changed to button with text

File: src/components/media-player/ha-media-player-browse.ts (hunk 2 above)
Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:731]: I don't see that here?
image

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: This is how it looks for me (Spotify Recently Played) in the "dev" branch:
image

Comment (ludeeus) [reply src/components/media-player/ha-media-player-browse.ts:731]: image
Mine looks correct (Spotify Recently Played)

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: It seems to be browser dependent. In Chrome the current dev coding looks good, but not in FF 80.0.1. When I apply my changes, then in Chrome the text moves even further away from the icon, so some margins seem to stack.

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: A quick glance shows, that this is already adding space between icon and title in Chrome. In FF this doesn't seem to be there:
image

Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:731]: What happens when you set that variable to 16px?

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: The whole red marked CSS block is not present in FF inspector, so it is not a matter of `--mdc-list-item-graphic-margin` being undefined, but it missing completely. The one below the red marked one, does exist in FF. The variables that are used there (e.g. `--mdc-list-item-graphic-size`) are not set in FF, but apparently in Chrome.

Comment (zsarnett) [reply src/components/media-player/ha-media-player-browse.ts:731]: Weird that this hasn't been brought up in the upstream repos...

Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:731]: OK, then set `--mdc-list-item-graphic-margin` to 0, and also check for RTL support.

File: src/components/media-player/ha-media-player-browse.ts [hunk 3]
```diff
@@ -726,6 +726,10 @@ export class HaMediaPlayerBrowse extends LitElement {
           background-color: transparent;
         }
 
+        mwc-list-item .title {
+          margin-left: 10px;
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:730]:
```suggestion
          margin-left: 16px;
```

File: src/panels/media-browser/ha-panel-media-browser.ts [hunk 4]
```diff
@@ -64,14 +64,14 @@ class PanelMediaBrowser extends LitElement {
...
-              <ha-svg-icon .path=${mdiPlayNetwork}></ha-svg-icon>
-            </mwc-icon-button>
+            <div>
+              <mwc-button @click=${this._showSelectMediaPlayerDialog}>
+                <ha-svg-icon .path=${mdiPlayNetwork}></ha-svg-icon>
+                ${this.hass.localize(
+                  "ui.components.media-browser.choose_player"
+                )}
+              </mwc-button>
+            </div>
```
Comment (bramkragten) [inline src/panels/media-browser/ha-panel-media-browser.ts:74]:
```suggestion
              <mwc-button @click=${this._showSelectMediaPlayerDialog}>
                <ha-svg-icon .path=${mdiPlayNetwork}></ha-svg-icon>
                ${this.hass.localize(
                  "ui.components.media-browser.choose_player"
                )}
              </mwc-button>
```

File: src/components/media-player/ha-media-player-browse.ts (hunk 2 above)
Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts:731]: Were any of you able to confirm my issue in FF on your instances?

Comment (SeanPM5) [reply src/components/media-player/ha-media-player-browse.ts]: Can confirm it's definitely an issue in Firefox.

Screen Shot 2020-09-08 at 12 01 34 PM

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts]: @bramkragten Added the fixes and verified RTL support. Let me know if I did it in the wrong way :smile: .

File: src/components/media-player/ha-media-player-browse.ts [hunk 5]
```diff
@@ -726,10 +728,14 @@ export class HaMediaPlayerBrowse extends LitElement {
...
         }
 
-        mwc-list-item .title {
+        mwc-list-item[dir="ltr"] .title {
           margin-left: 16px;
         }
 
+        mwc-list-item[dir="rtl"] .title {
+          margin-right: 16px;
+        }
```
Comment (zsarnett) [inline src/components/media-player/ha-media-player-browse.ts]:
```suggestion
        mwc-list-item .title {
          margin-left: 16px;
        }

        mwc-list-item[dir="rtl"] .title {
          margin-right: 16px;
          margin-left: 0;
        }
```

I think this is faster. Attributes are slow :)

Comment (spacegaier) [reply src/components/media-player/ha-media-player-browse.ts]: Will push a commit. Github somehow tells me your proposal is outdated, so I cannot accept it here directly.

Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts]: Do we set the dir attribute for this element?

Comment (zsarnett) [reply src/components/media-player/ha-media-player-browse.ts]: He added it yes

**Review by bramkragten — APPROVED** [review]
